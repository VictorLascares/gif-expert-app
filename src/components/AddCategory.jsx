import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories, categories }) => {
    const [category, setCategory] = useState('');

    const handleChange = (e) => {
        setCategory(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if( category.trim().length <= 1 ) return;
        if (categories.includes(category)) return;
        setCategories([ category, ...categories]);
        setCategory('');
    };

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={category}
                onChange={handleChange}
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
    categories: PropTypes.array.isRequired 
}