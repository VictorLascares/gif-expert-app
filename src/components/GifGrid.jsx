import { useFetchGifs } from "../hooks/useFetchGifs";
import { Spinner, GifItem } from "./";

export const GifGrid = ({ category }) => {
    const { gifs, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>

            <div className="card-grid">
                {isLoading ? <Spinner /> :
                    gifs.map(gif =>
                        <GifItem
                            key={gif.id}
                            {...gif}
                        />
                    )
                }
            </div>
        </>
    );
};