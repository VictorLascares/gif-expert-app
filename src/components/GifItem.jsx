const GifItem = ({ title, url }) => {
    return (
        <div className="card">
            <img src={url} alt={`Imagen de ${title}`} />
            <p>{title}</p>
        </div>
    );
};

export default GifItem;