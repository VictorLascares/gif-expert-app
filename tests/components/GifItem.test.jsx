import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Probando el componente <GifItem />', () => {
    const title = 'Dragon Ball Z';
    const url = 'https://dragon-ball-z.com';
    test('Evaluando el snapshot', () => {
        const { container } = render( 
            <GifItem 
                title={title} 
                url={url}
            />
        );

        expect(container).toMatchSnapshot();
    });
});