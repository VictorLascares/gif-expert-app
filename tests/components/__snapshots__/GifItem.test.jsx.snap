// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifItem /> Evaluando el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Imagen de Dragon Ball Z"
      src="https://dragon-ball-z.com"
    />
    <p>
      Dragon Ball Z
    </p>
  </div>
</div>
`;
